python/MANIFEST.in

```diff
@@ -16,7 +16,7 @@
 
 # Reference: https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html
 
-graft pyspark
+recursive-include pyspark *.pyi py.typed *.json
```

Comment:
@HyukjinKwon - Instead of this, I prefer to exclude tests as follows:

```diff
$ diff python/MANIFEST-json.in python/MANIFEST-notests.in 
19c19
< recursive-include pyspark *.pyi py.typed *.json
---
> graft pyspark
31a32
> global-exclude **/tests/**
```

That way, if we add new files in the future that don't happen to match the glob expressions on this line they will still be picked up.

But I believe you prefer the approach taken here, so I've started with that.